gradle-plugin/src/proguard/gradle/configuration/ProGuardTaskConfiguration.java

```diff
@@ -0,0 +1,475 @@
+    // Keep options for code.
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Optional @Input
+    public List<KeepClassSpecification> getKeep() {
+        return getConfiguration().keep;
+    }
+
+    @Optional @OutputFile
+    public File getPrintSeeds() {
+        return optionalFile(getConfiguration().printSeeds);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Shrinking options.
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Input
+    public boolean isShrink() {
+        return getConfiguration().shrink;
+    }
+
+    @Optional @OutputFile
+    public File getPrintUsage() {
+        return optionalFile(getConfiguration().printUsage);
+    }
+
+    @Optional @Input
+    public List<ClassSpecification> getWhyAreYouKeeping() {
+        return getConfiguration().whyAreYouKeeping;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Optimization options.
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Input
+    public boolean isOptimize() {
+        return getConfiguration().optimize;
+    }
+
+    @Optional @Input
+    public List<String> getOptimizations() {
+        return getConfiguration().optimizations;
+    }
+
+    @Input
+    public int getOptimizationPasses() {
+        return getConfiguration().optimizationPasses;
+    }
+
+    @Optional @Input
+    public List<ClassSpecification> getAssumeNoSideEffects() {
+        return getConfiguration().assumeNoSideEffects;
+    }
+
+    @Optional @Input
+    public List<ClassSpecification> getAssumeNoExternalSideEffects() {
+        return getConfiguration().assumeNoExternalSideEffects;
+    }
+
+    @Optional @Input
+    public List<ClassSpecification> getAssumeNoEscapingParameters() {
+        return getConfiguration().assumeNoEscapingParameters;
+    }
+
+    @Optional @Input
+    public List<ClassSpecification> getAssumeNoExternalReturnValues() {
+        return getConfiguration().assumeNoExternalReturnValues;
+    }
+
+    @Optional @Input
+    public List<ClassSpecification> getAssumeValues() {
+        return getConfiguration().assumeValues;
+    }
+
+    @Input
+    public boolean isAllowAccessModification() {
+        return getConfiguration().allowAccessModification;
+    }
+
+    @Input
+    public boolean isMergeInterfacesAggressively() {
+        return getConfiguration().mergeInterfacesAggressively;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Obfuscation options.
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Input
+    public boolean isObfuscate() {
+        return getConfiguration().obfuscate;
+    }
+
+    @Optional @OutputFile
+    public File getPrintMapping() {
+        return optionalFile(getConfiguration().printMapping);
+    }
```

Comment:
If `getTaskConfiguration()` is made public on `ProGuardTask`, this needs to return a provider. Otherwise with the code snippet posted above, configuring the publication will read the configuration eagerly.

I think this is true for the other optional output files as well.

Reply:
This doesn't need to be a provider. To depend on this output file, use `tasks.named('proguard').map {it.taskConfiguration.printMapping}`